src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -311,14 +323,75 @@ TEST_F(PolicyHandlerTest, OnPermissionsUpdated_method_With2Parameters) {
   // Check expectations
   EXPECT_CALL(app_manager_, application_by_policy_id(_))
       .WillOnce(Return(mock_app_));
-  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId_));
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
-              SendOnPermissionsChangeNotification(kAppId_, _, _));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
   // Act
   Permissions perms;
   policy_handler_.OnPermissionsUpdated(kPolicyAppId_, perms);
 }
 
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_TwoParams_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(_, _, _)).Times(0);
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions);
+}
+
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
+
```

Comment:
@dtrunov ,  there are two methods `OnPermissionsUpdated` in policy_handler. With two and three params. The second one calls first inside it's body. 
In test `OnPermissionsUpdated_TwoParams_InvalidApp_UNSUCCESS`  is calling func with two params and if `app` is invalid we don't  expect call SendOnPermissionsChangeNotification(_, _, _)).Times(0);
In second test `OnPermissionsUpdated_InvalidApp_UNSUCCESS` is calling func with three params, which contains calling func with two params and after this calling in test we expect invalid `app`, So in second test `EXPECT_CALL(mock_message_helper_, SendOnPermissionsChangeNotification(kAppId1_, _, _));` is from first func